src/core/WPE.js

```diff
@@ -66,10 +69,16 @@ export default class WPE {
     _noConnectionAfterTime() {
         console.log('_noConnectionAfterTime')
         if (this._state === this.STATES.NOIP) {
```

Comment:
I think we need to check if this is not at internet state, to fix the majority of the "rainy day connection" tests they had.

Reply:
Ok, so you are suggesting to use "this.STATES.HASINTERNET" or not ?

Reply:
I'd make this !== this.STATES.HASINTERNET, basically look for alternative or error whenever we do not have internet.